# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/media/MediaSpan.h
# Commit: 82c5bbd69b94
# Full Hash: 82c5bbd69b94982122235a17aa6dba1b6edcf55c
# Author: alwu <alwu@mozilla.com>
# Date: 2019-07-18 21:50:51
# Description:
#   Bug 1565501 - use fallible memory allocation in MediaSpan. r=jya
#   
#   Use fallible memory allocation in order to avoid the OOM issue.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D38400
# ==============================================================================

diff -r 0c4414e7ab91 -r 82c5bbd69b94 dom/media/MediaSpan.h
--- a/dom/media/MediaSpan.h	Thu Jul 18 00:05:57 2019 +0000
+++ b/dom/media/MediaSpan.h	Thu Jul 18 11:02:26 2019 +0000
@@ -78,15 +78,15 @@
       // append the new data into it.
       RefPtr<MediaByteBuffer> buffer =
           new MediaByteBuffer(mLength + aBuffer->Length());
-      if (!buffer->AppendElements(Elements(), Length()) ||
-          !buffer->AppendElements(*aBuffer)) {
+      if (!buffer->AppendElements(Elements(), Length(), fallible) ||
+          !buffer->AppendElements(*aBuffer, fallible)) {
         return false;
       }
       mBuffer = buffer;
       mLength += aBuffer->Length();
       return true;
     }
-    if (!mBuffer->AppendElements(*aBuffer)) {
+    if (!mBuffer->AppendElements(*aBuffer, fallible)) {
       return false;
     }
     mLength += aBuffer->Length();
